src/sbvr-compiler/odata-metadata-generator.ts

```diff
@@ -0,0 +1,173 @@
+import * as _ from 'lodash';
+import * as sbvrTypes from '@resin/sbvr-types';
+import { SqlModel, AbstractSqlTable } from '@resin/abstract-sql-compiler';
+
+// tslint:disable-next-line:no-var-requires
+const { version }: { version: string } = require('../../package.json');
+
+const getResourceName = (resourceName: string): string => {
+	const idParts = resourceName.split('-');
+
+	return idParts.map(idPart => idPart.split(/[ -]/).join('_')).join('__');
```

Comment:
Can `idPart` still contain a '-' after the first split above?

Reply:
It can't, I've updated the code